components/ArticleCategories/CategoryOption.js

```diff
@@ -84,29 +85,13 @@ const useStyles = makeStyles(theme => ({
   },
 }));
 
-const ArticleCategoryData = gql`
```

Comment:
user id is used to identify whether the current user added the category as well (so that they can remove those categories added by themselves)(in line 298), `disableVote` may be a bit confusing.
it's good to extract the author check logic to upper component though

Reply:
as for the mutation part it's certainly difficult to put them in appropriate place, `useMutation` is a hook and it uses closure to get required params, this means that we should put the params and `useMutation` in the same function, and at the same time consist the calling order so that it won't  break hook rules